src/app/server-cluster/ServerClusterInterface.h

```diff
@@ -0,0 +1,207 @@
+    //   if incrementing would exceed its maximum value. A cluster data version
+    //   SHALL be maintained for each cluster instance.
+    //   [...]
+    //   A cluster data version SHALL be incremented if any attribute data changes.
+    //
+    [[nodiscard]] virtual DataVersion GetDataVersion() const = 0;
+
+    [[nodiscard]] virtual BitFlags<DataModel::ClusterQualityFlags> GetClusterFlags() const = 0;
+
+    ///////////////////////////////////// Attribute Support ////////////////////////////////////////////////////////
+
+    /// ReadAttribute MUST be done on an "existent" attribute path: only on attributes that are
+    /// returned in an `Attributes` call for this cluster.
+    ///
+    /// `request.path` is expected to have `GetClusterId` as the cluster id as well as an attribute that is
+    /// included in a `Attributes` call.
+    ///
+    /// This MUST HANDLE the following global attributes:
+    ///   - FeatureMap::Id
+    ///   - ClusterRevision::Id
+    ///
+    /// This function WILL NOT be called for attributes that can be built out of cluster metadata.
+    /// Specifically this WILL NOT be called (and does not need to implement handling for) the
+    /// following attribute IDs:
+    ///     - AcceptedCommandList::Id
+    ///     - AttributeList::Id
+    ///     - GeneratedCommandList::Id
+    virtual DataModel::ActionReturnStatus ReadAttribute(const DataModel::ReadAttributeRequest & request,
+                                                        AttributeValueEncoder & encoder) = 0;
+
+    /// WriteAttribute MUST be done on an "existent" attribute path: only on attributes that are
+    /// returned in an `Attributes` call for this cluster.
+    ///
+    /// `request.path` is expected to have `GetClusterId` as the cluster id as well as an attribute that is
+    /// included in a `Attributes` call.
+    virtual DataModel::ActionReturnStatus WriteAttribute(const DataModel::WriteAttributeRequest & request,
+                                                         AttributeValueDecoder & decoder) = 0;
+
+    /// Attribute list MUST contain global attributes.
+    ///
+    /// Specifically these attributes MUST always exist in the list for all clusters:
+    ///     - ClusterRevision::Id
+    ///     - FeatureMap::Id
+    ///     - AcceptedCommandList::Id
+    ///     - AttributeList::Id
+    ///     - GeneratedCommandList::Id
+    /// See SPEC 7.13 Global Elements: `Global Attributes` table
+    ///
+    virtual CHIP_ERROR Attributes(const ConcreteClusterPath & path,
+                                  DataModel::ListBuilder<DataModel::AttributeEntry> & builder) = 0;
+
+    ///////////////////////////////////// Command Support /////////////////////////////////////////////////////////
+
+    virtual std::optional<DataModel::ActionReturnStatus>
+    InvokeCommand(const DataModel::InvokeRequest & request, chip::TLV::TLVReader & input_arguments, CommandHandler * handler) = 0;
```

Comment:
Please document the args.

Reply:
Copied over and adapted the DataModel::Provider documentation.